src/App.tsx

```diff
@@ -1,19 +1,171 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
+import 'bulma/css/bulma.min.css';
 
 import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './components/TodoList';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const getUserById = (userId: number) => (
+  users.find(user => user.id === userId)
+);
+
+const preparedTodos: Todo[] = todos.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId) || null,
+}));
+
+type State = {
+  todoList: Todo[];
+  newTitle: string;
+  selectedUserId: number;
+  isTitleInvalid: boolean;
+  isUserInvalid: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    todoList: [...preparedTodos],
+    newTitle: '',
+    selectedUserId: 0,
+    isTitleInvalid: false,
+    isUserInvalid: false,
+  };
+
+  handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      newTitle: event.target.value,
+      isTitleInvalid: false,
+    });
+  };
+
+  handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      selectedUserId: +event.target.value,
+      isUserInvalid: false,
+    });
+  };
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isFormValid = this.validateForm();
+
+    if (isFormValid) {
+      this.addNewTodo();
+      this.clearState();
+    }
+  };
+
+  validateForm = () => {
+    const { selectedUserId, newTitle } = this.state;
+
+    if (!selectedUserId || !newTitle.trim()) {
+      this.setState({
+        isTitleInvalid: !newTitle.trim(),
+        isUserInvalid: !selectedUserId,
+      });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  getNewTodo = () => {
+    const {
+      selectedUserId,
+      newTitle,
+      todoList,
+    } = this.state;
+
+    const newTodo = {
+      user: getUserById(selectedUserId) || null,
+      userId: selectedUserId,
+      id: todoList.length + 1,
```

Comment:
don't use length for id
![Screenshot 2022-01-31 at 13 45 04](https://user-images.githubusercontent.com/37452763/151788250-04d5c453-80cb-49eb-9793-a369b3ff613c.png)